frame/name-service/src/benchmarking.rs

```diff
@@ -0,0 +1,498 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Benchmarks for the name service pallet.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::{types::*, *};
+use crate::Pallet as NameService;
+use frame_benchmarking::{account, benchmarks, whitelisted_caller};
+use frame_support::traits::{Currency, Get, ReservableCurrency};
+use frame_system::{pallet_prelude::BlockNumberFor, Pallet as System, RawOrigin};
+use sp_io::hashing::blake2_256;
+use sp_runtime::traits::{Bounded, One};
+use sp_std::vec;
+
+type CurrencyOf<T> = <T as Config>::Currency;
+
+fn safe_mint<T: Config>() -> BalanceOf<T> {
+	CommitmentDeposit::<T>::get().unwrap() * 10000u32.into()
+}
+
+fn run_to_block<T: Config>(n: BlockNumberFor<T>) {
+	while System::<T>::block_number() < n {
+		NameService::<T>::on_finalize(System::<T>::block_number());
+		System::<T>::set_block_number(System::<T>::block_number() + One::one());
+		NameService::<T>::on_initialize(System::<T>::block_number());
+	}
+}
+
+fn register_new_para<T: Config>() -> (BoundedSuffixOf<T>, u32) {
+	let suffix: BoundedVec<u8, _> = BoundedVec::try_from("1234".as_bytes().to_vec()).unwrap();
+	let para_id = u32::max_value();
+	DomainRegistrations::<T>::insert(para_id, suffix.clone());
+	ReverseDomainsLookup::<T>::insert(suffix.clone(), para_id);
+	(suffix, para_id)
+}
+
+fn register_name_hash<T: Config>(
+	owner: T::AccountId,
+	name: Vec<u8>,
+	do_reveal: bool,
+) -> (NameHash, T::AccountId, T::AccountId) {
+	let caller = whitelisted_caller();
+	let secret = 3_u64;
+	T::Currency::make_free_balance_be(&caller, safe_mint::<T>());
+	T::Currency::make_free_balance_be(&owner, safe_mint::<T>());
+
+	let commitment_hash: CommitmentHash = NameService::<T>::commitment_hash(&name, secret.clone());
+	let origin = RawOrigin::Signed(caller.clone());
+
+	let _ = NameService::<T>::commit(origin.clone().into(), owner.clone(), commitment_hash.clone())
+		.expect("Commit succeeds.");
+
+	if do_reveal == true {
+		run_to_block::<T>(System::<T>::block_number() + 100u32.into());
+		let _ = NameService::<T>::reveal(origin.into(), name.clone(), secret, 100u32.into())
+			.expect("Reveal succeeds");
+	}
+
+	(NameService::<T>::name_hash(&name), owner, caller)
+}
+
+benchmarks! {
```

Comment:
benchmarks v2?

Reply:
👀 Are you referring to usage of the `#[benchmark]` maco?